sale_order_invoice_qty_number/models/sale_order.py

```diff
@@ -0,0 +1,14 @@
+# © 2023 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/lgpl-3.0.html
+from odoo import models, fields, api
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    invoice_ids = fields.Many2many("account.move")
+    invoice_number_count = fields.Integer(compute= '_compute_invoice_number_count', string='Invoice count', store= True)
+
+    @api.depends('order_line.invoice_lines')
+    def _compute_invoice_number_count(self):
+        for invoice in self:
+            invoice.invoice_number_count = len(invoice.invoice_ids)
```

Comment:
This is wrong: `invoice_number_count` computation should depend on `@api.depends` fields you've defined. In fact, we cannot use `invoice_ids`, because is a non-stored field (https://github.com/odoo/odoo/blob/15.0/addons/sale/models/sale_order.py#L228).

You need to obtain the invoice count using `order_line.invoice_lines`.